### PR TITLE
Add CLI tests for activation key delete

### LINKED ISSUE #1705: Add CLI test for activation-key delete
```
hammer activation-key delete --help

Usage:
    hammer activation-key delete [OPTIONS]

Options:
 --id ID                                 ID of the activation key
 --name NAME                             Activation key name to search by
 --organization ORGANIZATION_NAME        Organization name to search by
 --organization-id ORGANIZATION_ID       organization ID
 --organization-label ORGANIZATION_LABEL Organization label to search by
 -h, --help                              print help
```

### PR DESCRIPTION
**Pull request by oshtaier**
Adding coverage for activation key delete using different
approaches
Closes #1705

### CODE REVIEW

**Comment by JacobCallahan** [comment]
These methods should be split into separate tests. That way the entire test doesn't fail and our results are more representative. Also, delete by name can be removed, as per our discussion in #2163.

**Comment by JacobCallahan** [comment]
Actually, we should perform a better verification that the activation key was deleted.
Checking the result of info should work well.
```result = ActivationKey.info({'id': activation_key_id})```

**Comment by sthirugn** [comment]
I agree with @JacobCallahan's comments too.  ACK pending comments.

**Comment by elyezer** [comment]
+1 on creating three tests:

1. Delete Activation key using organization name
2. Delete Activation key using organization label
3. Delete Activation key using activation key name

**Comment by oshtaier** [comment]
Completely re-implemented all methods that we have for activation key delete functionality. Removed all stubs and added extra coverage that was required in them.

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key
..........
----------------------------------------------------------------------
Ran 10 tests in 219.062s

OK
```

File: tests/foreman/cli/test_activationkey.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def test_negative_create_activation_key_3(self):
         gen_string('utf8'),
         gen_string('html'),
     )
-    def test_positive_delete_activation_key_1(self, name):
+    def test_delete_activation_key_by_name(self, name):
```
Comment (elyezer) [inline tests/foreman/cli/test_activationkey.py]: :+1:

**Comment by elyezer** [comment]
I think that for tests where are being used values like organization name should e used activation key name because just the ID is enough when deleting the activation key (`ActivationKey.delete({'id': activation_key_id})) and it could mask other behaviours.

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key_by_org_name
.
----------------------------------------------------------------------
Ran 1 test in 50.046s

OK

nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key_by_org_label
.
----------------------------------------------------------------------
Ran 1 test in 48.110s

OK
```

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_activationkey.py -m test_delete_activation_key
..........
----------------------------------------------------------------------
Ran 10 tests in 221.699s

OK
```

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
Third ACK.

**Pull request merged**